### DISCUSSION TITLE
College Baseball and Softball

### DISCUSSION THREAD

**Discussion by NRutt2005** [vasqued2/ha-teamtracker#78] [Ideas]
Hello,

I was wondering if there is anyway to get the College Baseball and College Softball integrated into the ha-teamtracker integration? I found the ESPN API for both sports. However, I cannot figure out how to get these two sports to integrate with the team-tracker. Below are the ESPN API url's.

College Baseball:
https://site.api.espn.com/apis/site/v2/sports/baseball/college-baseball/scoreboard

College Softball:
https://site.api.espn.com/apis/site/v2/sports/baseball/college-softball/scoreboard

Thank you!

**Comment by vasqued2** [comment]
Custom API configuration should work.  Are there issues with it?

[https://github.com/vasqued2/ha-teamtracker/wiki/Custom-API-Configurations](https://github.com/vasqued2/ha-teamtracker/wiki/Custom-API-Configurations)

**Reply by NRutt2005** [reply]
I've tried the custom API configuration but neither the softball or baseball is pulling data. I've attached a screen shot of the two cards and then below is the code lines for the custom API configuration. Let me know if you see anything wrong in my formatting. Thanks!

https://pastebin.com/v0beJb1Q

<img width="1235" alt="Screenshot 2023-03-12 at 3 01 13 PM" src="https://user-images.githubusercontent.com/75466468/224570197-91fcdcc4-47b3-4509-9db5-3f80560ebcf0.png">

**Comment by vasqued2** [comment]
League_id should be XXX

team_id should be OK or OKL or whatever abbreviation ESPN uses, not a number.

**Reply by NRutt2005** [reply]
@vasqued2 According to the data on the JSON data provided on the API site, the team abbreviation is "OU". I've tried OU and I've tried "Oklahoma", "OK", and "OKL". Nothing seems to be working. The ESPN website shows the abbreviation as "OU" as well (**_screenshot below_**). So my thinking is that OU is what I need to use. Am I incorrect on the league_id being 102 (**_screenshot below_**)? I've attached screenshots to help to see what I'm seeing. Thank you for all your help. Hopefully I can get this working.

https://pastebin.com/nwLqSY9e

![SoftballBaseball-HomeAssistantPage](https://user-images.githubusercontent.com/75466468/225062354-ef6f8dd9-630a-4550-889c-ce53db430193.PNG)

![ESPN-TeamAbbreviation](https://user-images.githubusercontent.com/75466468/225064531-642504c5-8df2-4f53-bba5-0c980f2ec352.PNG)

![JSON-CollegeSoftball-LeagueID](https://user-images.githubusercontent.com/75466468/225065649-9dcc6fc6-7705-4343-b322-c94c09b89588.PNG)

**Comment by vasqued2** [comment]
102 is incorrect.  It should literally be XXX as I stated above.  Not a number.

**Comment by NRutt2005** [comment]
My apologies for misinterpreting this "XXX" part. I have seen "xxx" used as a way to start this is a variable that is determined by the whatever data you are trying to see (the unknown variable). Since I switched this over to "xxx" I was able to see the data for about 2 hours this morning for Softball only then the data went back to "NOT_FOUND" state.  

When in the PRE state I was able to see the data in the card as expected. I'm not sure what is cause this to go from PRE to NOT_FOUND state...I'm not requesting very many teams from the API total. 4 through the team tracker GUI integration and then the two baseball and softball custom integrations. I am having this issue with other teams that I have tested with the custom integration method. I've attached more screenshots for you. The custom integration showed up for a little bit while drafting this message earlier today. It has since gone into the "NOT_FOUND" state again.

![SoftballShowing](https://user-images.githubusercontent.com/75466468/225803411-8b62850c-57fc-4aec-a2e8-65ee3c2c36ac.PNG)
![SoftballShowing-StateHistory](https://user-images.githubusercontent.com/75466468/225803416-20d661e7-4920-47b3-aec3-e81e9ee9f649.PNG)

**Comment by vasqued2** [comment]
You are probably bumping up,against the default API limit of 25 games.  You can change it in the const.py file or wait for me to make a change for baseball in the next release that will try to fix it.  By default, ESPN limits to only return 25 games at a time.  Since there are more than 25 games yesterday and today, it truncates the result.  Baseball seems to be the major sport the routinely hits the limits.  Most other sports/leagues don’t have 25 games back to back.

if you look at the API message on the sensor or look at the log, you will probably see the warning message.

**Reply by NRutt2005** [reply]
I was able to modify the const.py file API limit from 25 to 75 and this fixed my issues completely.

![image](https://user-images.githubusercontent.com/75466468/226467457-ac3399dc-3fb7-4172-a9c2-a78803318f3a.png)

**Comment by NRutt2005** [comment]
Just a small question does the is the refresh time for the custom API integrations different than the non-custom integrations? Example, the NBA, for NCAA Football API integrations refresh almost instantaneously while the baseball/softball custom integrations are quite delayed. Is it just depending on how often ESPN has the API refresh or is this a setting that can be modified in the ha-teamtracker config files? Thanks for all the help with getting the custom API integration working correctly for these not as popular sports.

**Comment by vasqued2** [comment]
It's entirely dependent upon the ESPN API.  It pulls the data at the same frequency (5 seconds) for all APIs while the games are in progress.  ESPN seems to update the info at different frequencies based on sport and leagues.
